Lib/concurrent/futures/thread.py

```diff
@@ -96,8 +98,12 @@ def __init__(self, max_workers=None, thread_name_prefix=''):
         if max_workers <= 0:
             raise ValueError("max_workers must be greater than 0")
 
+        # check that max_queue_size is a positive integer
+        if type(max_queue_size) is not int or max_queue_size < 0:
```

Comment:
Should it strictly be an `int`? Classes that inherit from `int` won't pass this condition unless `isinstance` is used. Either way I'm not sure if doing any check for the type is a good idea since the `Queue` class generally doesn't (and fails when you `put` something in it)